scalameta/parsers/shared/src/main/scala/scala/meta/internal/parsers/ScalametaParser.scala

```diff
@@ -2957,6 +2990,16 @@ class ScalametaParser(input: Input, dialect: Dialect) { parser =>
     Defn.Object(mods, termName(), templateOpt(OwnedByObject))
   }
 
+  def enumDef(mods: List[Mod]): Defn.Enum = atPos(mods, auto) {
+    accept[KwEnum]
+    rejectMod[Mod.Override](mods, Messages.InvalidOverrideEnum)
```

Comment:
This is partially broken in Dotty, but the only valid modifiers for enums are annotations and access modifier (`private`, `protected`). For enum cases, only annotations should be valid.

Can you add tests cases as well

Reply:
Ok I'll change that. So an enum cannot be final ?

Reply:
No, it is redundant. In `enum Foo { case Bar }`, you should never be able to extends `Foo` or `Bar`.

It is like `object`s. They are implicitly `final`. Although you can write `final object Foo` (it should probably be flagged as an error)

Reply:
Oh I see, thank you !